Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Weights.h

```diff
@@ -64,8 +64,8 @@ struct Cotangent_value_Meyer_impl
     if(divider == 0 /*|| divider != divider*/) 
     {
       CGAL::collinear(get(ppmap, v0), get(ppmap, v1), get(ppmap, v2)) ? 
-        CGAL_warning(!"Infinite Cotangent value with degenerate triangle!") :
-        CGAL_warning(!"Infinite Cotangent value due to floating point arithmetic!");
+        CGAL_warning_msg(false, "Infinite Cotangent value with degenerate triangle!") :
+        CGAL_warning_msg(false, "Infinite Cotangent value due to floating point arithmetic!");
```

Comment:
How can that compile?!

Reply:
... With `CGAL_NDEBUG`, `CGAL_warning_msg(x)` evaluates to `(static_cast<void>(0))`, and the ternary operator becomes a valid expression of type `void`.